src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,244 @@
+                    return arrowExpressionNodes != null;
+                }
+            }
+
+            return false;
+        }
+
+        protected override IParameterSymbol? GetParameterSymbol(SemanticModel semanticModel,ArgumentSyntax argumentSyntaxNode, CancellationToken cancellationToken)
+            => argumentSyntaxNode.DetermineParameter(semanticModel, allowParams: true, cancellationToken);
+
+        protected override SyntaxNode? GetInlineStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            var declarationSyntax = (MethodDeclarationSyntax)calleeMethodDeclarationSyntaxNode;
+            SyntaxNode? inlineSyntaxNode = null;
+            var blockSyntaxNode = declarationSyntax.Body;
+            // 1. If it is a ordinary method with block
+            if (blockSyntaxNode != null)
+            {
+                var blockStatements = blockSyntaxNode.Statements;
+                if (blockStatements.Count == 1)
+                {
+                    inlineSyntaxNode = GetExpressionFromStatementSyntaxNode(blockStatements[0]);
+                }
+
+                return inlineSyntaxNode;
+            }
+            else
+            {
+                // 2. If it is using Arrow Expression
+                var arrowExpressionNode = declarationSyntax.ExpressionBody;
+                if (arrowExpressionNode != null)
+                {
+                    inlineSyntaxNode = arrowExpressionNode.Expression;
+                }
+
+                return inlineSyntaxNode;
+            }
+
+            // A check has been done before to make sure there is one arrow expression or block statement.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        private static SyntaxNode? GetExpressionFromStatementSyntaxNode(StatementSyntax statementSyntax)
+            => statementSyntax switch
+            {
+                ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                _ => null
+            };
+
+        protected override bool ShouldCheckTheExpressionPrecedenceInCallee(SyntaxNode syntaxNode)
+            => s_syntaxKindsNeedsToCheckThePrecedence.Any(syntaxNode.IsKind);
+
+        protected override bool NeedWrapInParenthesisWhenPrecedenceAreEqual(SyntaxNode calleeInvocationSyntaxNode)
+        {
+            var parent = calleeInvocationSyntaxNode.Parent;
+            if (parent != null)
+            {
+                // For left associative expression
+                // If in the original invocation, it is the left child. Since it is a left associative expression,
+                // then it is safe to replace it without parenthesis.
+                // Example:
+                // int Callee(int i, int j) => i + j;
+                // void Caller()
+                // {
+                //   var a = F(1, 2) - 3 - 2 - 1;
+                // }
```

Comment:
i don't get this example.  `Callee` is not called from `Caller`.

Reply:
This method is removed so I'll mark it as resolved.